### PR TITLE
Clarify "Update from v3.3.x to v3.3.latest"

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [CS-10132](https://jsd.ez.no/browse/CS-10132)
| Versions      | 3.x, 4.x

TODO:
- [ ] Composer must be 2.1+; Already indicated in https://doc.ibexa.co/en/3.3/getting_started/requirements/#package-manager, could it be useful to say it again here?
- [ ] ~~`composer require symfony/workflow ^5.4 --no-scripts` before requiring the last 3.3.x to avoid `ezsystems/ezplatform-workflow[v2.3.10, ..., v2.3.14] require symfony/workflow ^5.4 -> found symfony/workflow[v5.4.0, ..., v5.4.9] but these were not loaded, likely because it conflicts with another require.` (≤ v3.3.12); can be replaced by the whole Symfony update.~~ [Not understood enough to be treated this time.]
- [x] _Update Flex server_: The Flex server must be updated before requiring the last 3.3.x
- [x] _Update Flex server_: symfony/flex must be updated to properly use the new endpoint and avoid errors like `The "https://api.github.com/repos/ibexa/recipes/contents/index.json?ref=flex/main/aliases.json" file could not be downloaded` (v3.3.8 comes with symfony/flex v1.15.3)
- [ ] ~~_Update Flex server_: For 3.3.8 and later, not really need to run `composer recipes:install` BUT for 3.3.7 and earlier, this is needed for hautelook/templated-uri-bundle replacement.~~ [Sometime needed, sometime harmlessly useless, just keep it.]
- [ ] ~~_Update Flex server_: no need to run `composer run post-install-cmd`~~ [Harmless]
- [x] _v3.3.2_ mysql CLI: no space allowed between `-p` and password.
- [x] _v3.3.7_ `Migration "src/bundle/Resources/install/migrations/content/Components/move_configuration_to_settings.yaml" not found.`
- [x] _v3.3.7_ there is a pair of ibexa-3.3.6-to-3.3.7.sql for mysql and postgresql
- [x] _v3.3.9_ mysql CLI: no space allowed between `-p` and password.
- [x] _3.3.15_: To replace SF 5.3 with 5.4 in composer.json isn't enough, a command like `composer update "symfony/*"` must also be run.
- [x] _3.3.15_: To replace SF 5.3 with 5.4 must be done on 3.3.12 and earlier, 3.3.13 and above are already on 5.4
- [x] _3.3.15_: `Unrecognized option "timeout" under "nelmio_solarium.endpoints.default"` during SF update.
- [x] _3.3.15_: command from manifest.json were here earlier than 3.3.15, they are in 3.3.8 but the order changed.
- [x] _3.3.15_ → _v3.3.15_ like other sections

### CODE REVIEW

**Comment by adriendupuis** [comment]
`composer require symfony/workflow ^5.4` must be run before `composer require ibexa/experience:3.3.21` on 3.3.12 and below; 3.3.13 and above doesn't need it.

**Comment by adriendupuis** [comment]
See https://github.com/adriendupuis/ezplatform-ee-docker/pull/70#issuecomment-1217963548 for a script testing updates for each version/patch of 3.3 in each flavor.

File: docs/update_and_migration/from_3.3/update_from_3.3.md [hunk 1]
```diff
@@ -284,16 +267,43 @@ Run the following scripts:
...
     ```
 
+### v3.3.13
+
+!!! note "Symfony 5.4"
+
+    Prior to 3.3.13, Symfony 5.3 was used by default.
+
+    If you are still using Symfony 5.3, you need to update your installation to Symfony 5.4.
+    To do this, update your composer.json to refer to `5.4.*` instead or `5.3.*`.
```
Comment (DominikaK) [inline docs/update_and_migration/from_3.3/update_from_3.3.md:287]:
```suggestion
    To do this, update your `composer.json` to refer to `5.4.*` instead or `5.3.*`.
```

File: docs/update_and_migration/from_3.3/update_from_3.3.md [hunk 2]
```diff
@@ -284,16 +267,43 @@ Run the following scripts:
...
+
+    Refer to the relevant website skeleton for an example: [content](https://github.com/ibexa/content-skeleton/blob/v3.3.13/composer.json), [experience](https://github.com/ibexa/experience-skeleton/blob/v3.3.13/composer.json), [commerce](https://github.com/ibexa/commerce-skeleton/blob/v3.3.13/composer.json).
+
+    The following `sed` commands should update the right lines, use with caution and properly check the result:
```
Comment (DominikaK) [inline docs/update_and_migration/from_3.3/update_from_3.3.md:291]:
```suggestion
    The following `sed` commands should update the relevant lines.
    Use them with caution and properly check the result:
```

File: docs/update_and_migration/from_3.3/update_from_3.3.md [hunk 3]
```diff
@@ -284,16 +267,43 @@ Run the following scripts:
...
+
+    ```shell
+    sed -i -E 's/"symfony\/(.+)": "5.3.*"/"symfony\/\1": "5.4.*"/' composer.json;
+    sed -i -E 's/"require": "5.3.*"/"require": "5.4.*"/' composer.json;
+    ```
+
+    After this composer.json update, run `composer update "symfony/*"`.
```
Comment (DominikaK) [inline docs/update_and_migration/from_3.3/update_from_3.3.md:298]:
```suggestion
    After this `composer.json` update, run `composer update "symfony/*"`.
```

**Review by DominikaK — APPROVED** [review]

File: docs/update_and_migration/from_3.3/update_from_3.3.md [hunk 4]
```diff
@@ -284,16 +267,44 @@ Run the following scripts:
...
+    ```
+
+    After this `composer.json` update, run `composer update "symfony/*"`.
+
+    You may have to adapt some configuration to fit this new Symfony minor version.
+    For example, you might have to remove `timeout` related config from `nelmio_solarium` bundle config:
+    ```shell
+    sed -i -E '/ *timeout: [0-9]+/d' ./config/packages/nelmio_solarium.yaml ./config/packages/ezcommerce/ezcommerce_advanced.yaml
+    composer update "symfony/*"
+    ```
```
Comment (MagdalenaZuba) [inline docs/update_and_migration/from_3.3/update_from_3.3.md:306]:
```suggestion
    For example, you might have to remove `timeout` related config from `nelmio_solarium` bundle config:
    
    ```shell
    sed -i -E '/ *timeout: [0-9]+/d' ./config/packages/nelmio_solarium.yaml ./config/packages/ezcommerce/ezcommerce_advanced.yaml
    composer update "symfony/*"
    ```
```

File: docs/update_and_migration/from_3.3/update_from_3.3.md [hunk 5]
```diff
@@ -284,16 +267,44 @@ Run the following scripts:
...
+    Use them with caution and properly check the result:
+
+    ```shell
+    sed -i -E 's/"symfony\/(.+)": "5.3.*"/"symfony\/\1": "5.4.*"/' composer.json;
+    sed -i -E 's/"require": "5.3.*"/"require": "5.4.*"/' composer.json;
+    ```
+
+    After this `composer.json` update, run `composer update "symfony/*"`.
+
+    You may have to adapt some configuration to fit this new Symfony minor version.
```
Comment (MagdalenaZuba) [inline docs/update_and_migration/from_3.3/update_from_3.3.md:301]:
```suggestion
    You may need to adapt configuration to fit the new minor version of Symfony.
```

**Review by MagdalenaZuba — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved.

**Pull request merged**